src/utils/index.ts

```diff
@@ -1,8 +1,10 @@
-export const isObject = (obj: any): boolean => {
+import { JSONValue, JSONArray, JSONObject } from '../';
+
+export const isObject = (obj: JSONValue): boolean => {
```

Comment:
I can see in a few places you using this like so
`if (isObject(a)) { doThing(a as JSONObject)`

You can actually upgrade this function to tell the compiler this, so you don't need to cast.
```suggestion
export const isObject = (obj: JSONValue): obj is Object => {
```
Now when used in the same context the type checker will know that inside the if block the parameter is an object. You can cast to any type like this, the only requirement being that the function return a boolean.

Reply:
Additionally you may want to consider changing the parameter type to "unknown", it will probably make the function easier to use. Although this obviously depends on the guarantees the method gives.

Reply:
Done